util/buildflags/utils.go

```diff
@@ -0,0 +1,54 @@
+package buildflags
+
+import (
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/gocty"
+)
+
+type comparable[E any] interface {
+	Equal(other E) bool
+}
+
+func removeDupes[E comparable[E]](s []E) []E {
```

Comment:
This seems to be duplicated. Additionally (not for this PR) there is also `removeDupesStr` but it uses map. I think this could also avoid N^2 complexity but not sure if worth it as the inputs are likely small.

Reply:
The purpose of this was it could utilize the `Equal` method that was added so it wasn't so dependent on strings. Due to the small number of inputs, the N^2 complexity isn't likely to be an issue. This might actually be faster than `removeDupesStr` just because maps have some issues when used for deduplicating small inputs due to overhead from memory allocation and hash key lookups.

This was duplicated and I've removed the duplicate from `bake/bake.go`. When I originally did this, it was in `bake/bake.go` and then I started moving stuff. It was easier to just copy it at the time. Since I missed the change to call `Normalize()` for `t.SSH` as pointed out in the other comment, the linter didn't catch that the original call was no longer needed since all calls had been moved to this file. I've changed that to use `Normalize()` and removed the duplicate.